### PR TITLE
doc: kconfig: Add some docs on 'if' and sanity checking in menuconfig

### PR DESCRIPTION
**Pull request by ulfalizer**
Two changes to the Kconfig best practices page:

```
doc: kconfig: Add section on (lack of) conditional includes

Add a section on how 'if' works to the 'Kconfig - Tips and Best
Practices' page, especially when combined with 'source'.

There seems to have been some confusion here that lead to a bunch of
duplicated dependencies. It's probably simpler than most people assume.
```

```
doc: kconfig: Encourage using menuconfig as a sanity check

Some people never use the menuconfig, but it makes a good sanity check
when making Kconfig changes, so encourage it for that. Mention a few
things that can be checked in it.
```

### CODE REVIEW

File: doc/guides/kconfig/index.rst [hunk 1]
```diff
@@ -275,6 +275,85 @@ In summary, here are some recommended practices for ``select``:
   you like. They're a great tool for simplifying Kconfig files.
 
 
+(Lack of) conditional includes
+******************************
+
+``if`` blocks add dependencies to each item within the ``if``, as if via
```
Comment (aescolar) [inline doc/guides/kconfig/index.rst]: Minor thing: The "as if via" construct seems a bit difficult to read.
Maybe something like "as if depends on was used" would be a bit better?

**Review by aescolar — APPROVED** [review]
Thanks :)

File: doc/guides/kconfig/index.rst (hunk 1 above)
Comment (ulfalizer) [reply doc/guides/kconfig/index.rst]: Yup, better. Changed.

File: doc/guides/kconfig/index.rst [hunk 2]
```diff
@@ -275,6 +275,85 @@ In summary, here are some recommended practices for ``select``:
   you like. They're a great tool for simplifying Kconfig files.
 
 
+(Lack of) conditional includes
+******************************
+
+``if`` blocks add dependencies to each item within the ``if``, as if ``depends
+on`` was used.
+
+A common misunderstanding related to ``if`` is to think that this code
```
Comment (dbkinder) [inline doc/guides/kconfig/index.rst]:
```suggestion
A common misunderstanding related to ``if`` is to
think that the following code example
```

**Review by tyrantlizard77 — APPROVED** [review]

File: doc/guides/kconfig/index.rst [hunk 3]
```diff
@@ -275,6 +275,85 @@ In summary, here are some recommended practices for ``select``:
...
+**********************************
+
+When adding new symbols or making other changes to Kconfig files, it is a good
+idea to look up the symbols in the :ref:`menuconfig <override_kernel_conf>`
+interface afterwards. To get to a symbol quickly, use the menuconfig's jump-to
+feature (press :kbd:`/`).
+
+Here are some things to check:
+
+* Are the symbols placed in a good spot in the menu structure?
```
Comment (dbkinder) [inline doc/guides/kconfig/index.rst]: Can you explain what a "good spot" means?   Logical, alphabetized, ...?

File: doc/guides/kconfig/index.rst [hunk 4]
```diff
@@ -275,6 +275,85 @@ In summary, here are some recommended practices for ``select``:
...
+
+* Is it easy to guess what the symbols do from their prompts?
```
Comment (dbkinder) [inline doc/guides/kconfig/index.rst]:
```suggestion
* Is it easy to understand what the symbols do from their prompts?
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by dbkinder — CHANGES_REQUESTED** [review]
some suggested edits

File: doc/guides/kconfig/index.rst (hunk 2 above)
Comment (ulfalizer) [reply doc/guides/kconfig/index.rst]: Fixed

File: doc/guides/kconfig/index.rst (hunk 3 above)
Comment (ulfalizer) [reply doc/guides/kconfig/index.rst]: Clarified a bit and added some more details.

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks!

**Pull request merged**